### PR TITLE
Send ddb region as awsCfg

### PR DESCRIPTION
**Pull request by danielblando**
**What this PR does**:
Send DynamoDB region as awsConfig instead of session config

**Which issue(s) this PR fixes**:
Fixes #5840

**Checklist**
- [X] Tests updated
- [ ] Documentation added
- [X] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: pkg/ring/kv/dynamodb/dynamodb.go [hunk 1]
```diff
@@ -57,13 +57,12 @@ func newDynamodbKV(cfg Config, logger log.Logger) (dynamodbKV, error) {
 		return dynamodbKV{}, err
 	}
 
+	awsCfg := aws.NewConfig()
 	if len(cfg.Region) > 0 {
```
[the remark this answers is not in the document]
Comment (danielblando) [reply pkg/ring/kv/dynamodb/dynamodb.go:61]: Region here is optional in the config.   
I would need to do a test to see what the client will do if we send an empty string. I am not sure if it will try to use the env var.

[the remark this answers is not in the document]
Comment (yeya24) [reply pkg/ring/kv/dynamodb/dynamodb.go:61]: Cool just a nit

**Review by yeya24 — APPROVED** [review]
